### ISSUE TITLE
Handling assertion error

### ISSUE THREAD

**Issue by Bishalsahoo** [Tencent/rapidjson#1315]
Hello . I am working on a project in C++ lib where I am utilizing rapidjson . As the library is being utilized in multiple platforms , is there a way to handle the assertion error so that they can get information about the error rather than NULL . Any help will be appreciated.

**Comment by miloyip** [comment]
Assertions in C/C++ are mostly for enforcing valid calls in debug build.
They should not be used as runtime error checking.
If there is assertion failure, it must be that the code has incorrect logic.

**Comment by mavericksthinker** [comment]
I agree . But the intention of doing that is for other platforms if they provide input which is not according to the structure ex if in the code the key required is name but accidentally they sent it as names which will cause assertion false . They get result as null coz they are unable to know if that's error in name or the json string that they are sending is breaking or another error ,which leads them to the conclusion that my lib has error which has passed 400 unit tests on all possible scenario. Even when they provided me the input they were testing it was showed me assertion false so I was able to solve and got the output . But on Android,iOS, they get null . Even for php and nodejs they are getting null . I can manage those assertion error manually but handling all those is time consuming. So I wanted to catch the assertion as an exception and this it but as assertion can't be handle by that ,so I asked this question. As modifying the document.h is also kinda not effective yet.
